### PR TITLE
List fixes

### PR DESCRIPTION
**Pull request by blasten**
This should fix issues https://github.com/PolymerElements/iron-list/issues/214, https://github.com/PolymerElements/iron-list/issues/213 and make the code a little more clear.

cc @kevinpschaaf

### CODE REVIEW

File: iron-list.html [hunk 1]
```diff
@@ -593,10 +589,14 @@
     attached: function() {
       this.updateViewportBoundaries();
       this._render();
+      // `iron-resize` is fired when the list is attached if the event is added
+      // before attached causing unnecessary work.
+      this.listen(this, 'iron-resize', '_resizeHandler');
```
Comment (frankiefu) [inline iron-list.html]: How about just keep the listeners object and in `_resizeHandler` we add a check for `this.isAttached`?

**Comment by frankiefu** [comment]
When I tried the demo/index.html and resized the window, the header condensed and the scroll pos changed.

**Comment by frankiefu** [comment]
And on Safari, it's even worse.  The header condensing and expanding continuously while the window is resized.

File: iron-list.html (hunk 1 above)
Comment (blasten) [inline iron-list.html]: by the moment `iron-resize` fired, `isAttached` is true

**Comment by frankiefu** [comment]
LGTM

**Pull request merged**